packages/ai-jsx/src/core/completion.tsx

```diff
@@ -40,18 +42,50 @@ export type ModelComponent<T extends ModelPropsWithChildren> = Component<T>;
  */
 export interface FunctionDefinition {
   description?: string;
-  parameters: Record<string, FunctionParameter>;
+  parameters: FunctionParameters;
+}
+
+export function getParametersSchema(parameters: FunctionParameters) {
```

Comment:
We are allowing both the simplified function signature as well as the more complex `zod` Schema.